tests/5.2/runtime_calls/test_omp_in_explicit_task.c

```diff
@@ -0,0 +1,52 @@
+//===---- test_omp_in_explicit_task.c ----------------------------------------------===//
+// 
+// OpenMP API Version 5.2 Nov 2021
+//
+// Uses omp_in_explicit_task() runtime call to ensure it is 1 when in an explicit
+// task and 0 otherwise.
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ompvv.h"
+#include <stdbool.h>
+
+#define N 1024
+
+int errors;
+
+int test_wrapper() { 
+    errors = 0;
+    int A[N];
+    for(int i = 0; i < N; i++){
+        A[i] = 0;
+    }
+    #pragma omp task shared(A)// creates EXPLICIT task, omp_in_explicit_task = 1
+    {
+        for(int i = 0; i < N; i++){
+            A[i] = omp_in_explicit_task();
+        }
+    }
+    for(int i = 0; i < N; i++){
+        OMPVV_TEST_AND_SET(errors, A[i] == 0);
+    }
+    OMPVV_WARNING_IF(A[5] == 0, "omp_in_explicit_task() did not return correct value");
+    for(int i = 0; i < N; i++){
+        #pragma omp parallel for // creates IMPLICIT tasks, omp_in_explicit_task = 0
+        for(int i = 0; i < N; i++){
+            A[i] += omp_in_explicit_task();
+        }
+    }
+    for(int i = 0; i < N; i++){
+        OMPVV_TEST_AND_SET(errors, A[i] == 0);
```

Comment:
This test does not necessarily check the above implicit task test; a better way to check both explicit task test and implicit test will be the following:

1) Line 34 will check the explicit task test.
2) In line 40, `A[i] = omp_in_explicit_task();`
3) In line 44, `OMPVV_TEST_AND_SET(errors, A[i] != 0);` which will check the implicit task test

Reply:
correct, thank you!